### ISSUE TITLE
ORM redesign: add the new ORM classes as default loaded for `verdi shell`

### ISSUE THREAD

**Issue by sphuber** [aiidateam/aiida_core#2174]
The list of default modules is defined in `aiida.cmdline.utils.shell.DEFAULT_MODULES_LIST`

**Comment by giovannipizzi** [comment]
Prepare first a list of all things that should be loaded by default for discussion

**Comment by sphuber** [comment]
Suggested default list:
```
    ('aiida.orm', 'Node', 'Node'),
    ('aiida.orm', 'ProcessNode', 'ProcessNode'),
    ('aiida.orm', 'CalculationNode', 'CalculationNode'),
    ('aiida.orm', 'CalcJobNode', 'CalcJobNode'),
    ('aiida.orm', 'CalcFunctionNode', 'CalcFunctionNode'),
    ('aiida.orm', 'WorkflowNode', 'WorkflowNode'),
    ('aiida.orm', 'WorkChainNode', 'WorkChainNode'),
    ('aiida.orm', 'WorkFunctionNode', 'WorkFunctionNode'),
    ('aiida.orm', 'CalculationFactory', 'CalculationFactory'),
    ('aiida.orm', 'DataFactory', 'DataFactory'),
    ('aiida.orm', 'WorkflowFactory', 'WorkflowFactory'),
    ('aiida.orm', 'Data', 'Data'),
    ('aiida.orm', 'Bool', 'Bool'),
    ('aiida.orm', 'Float', 'Float'),
    ('aiida.orm', 'Int', 'Int'),
    ('aiida.orm', 'Str', 'Str'),
    ('aiida.orm', 'List', 'List'),
    ('aiida.orm', 'ParameterData', 'ParameterData'),
    ('aiida.orm', 'Code', 'Code'),
    ('aiida.orm', 'Computer', 'Computer'),
    ('aiida.orm', 'Group', 'Group'),
    ('aiida.orm', 'QueryBuilder', 'QueryBuilder'),
    ('aiida.orm', 'User', 'User'),
    ('aiida.orm', 'load_code', 'load_code'),
    ('aiida.orm', 'load_computer', 'load_computer'),
    ('aiida.orm', 'load_group', 'load_group'),
    ('aiida.orm', 'load_node', 'load_node'),
```
Basically all the base `Node` sub classes, base `Data` sub classes, some commonly used entity classes and the four entity loaders.

**Comment by CasperWA** [comment]
Is there a reason to not have all the `Data` sub-classes be imported, i.e. `StructureData`, `KpointsData`, etc.?

**Comment by sphuber** [comment]
The reasoning there is that we do not want to import everything as importing too much will slow things down. Here we only import the base data types as these are so generic that they will always be part of `aiida-core`. The classes like `StructureData` are already a lot more domain specific and will likely in the future be moved to a general plugin like `aiida-solid-state` or something like that. In addition, the `verdi config` command allows a user to customize this auto-import list on a per-profile basis, so one can add the data classes that are used often

**Comment by ltalirz** [comment]
To me the choice looks quite clean.

**Comment by sphuber** [comment]
Fixed in PR #2533
